### PR TITLE
Time-series query uses a sort on the chosen time column

### LINKED ISSUE #13341: Time Series Insight GUID issue
### Describe the Bug

Imported activity data from another source using GUID and the time series graph is quite entertaining. Looking at the api query, there is no sort option. I think its worth including the sort parameter using the chosen time column.

image

### To Reproduce

Create a table with a guid primary key and a date field then create a time series chart.

### Errors Shown

N/A

### What version of Directus are you using?

9.10.0

### What version of Node.js are you using?

16.13.1

### What database are you using?

MS SQL

### What browser are you using?

Chrome and MS Edge

### How are you deploying Directus?

locally

### PR DESCRIPTION
**Pull request by jamescammarano**
I believe this closes #13341

Sorts the x-axis by oldest to most recent. 

Note: the dataset below has dates in the future that are not shown on the graph because the graph filters on NOW.  This might be considered a limitation on time-series if someone wanted to do some sort of forecasting but that is out of the scope of this PR.

https://user-images.githubusercontent.com/67079013/170571956-9dcc9cb6-81f2-41b5-a1d7-c607e0a6a5e1.mov

Screen Shot 2022-05-26 at 4 11 17 PM

### CODE REVIEW

**Comment by SanthoshMR** [comment]
Has this been tested against any DBs? In Postgres and related DBs, this will throw an error since the grouping is done based on the precision (year, month, week...etc.,) and the dateField used in the sort query is not part of the group query.

**Comment by azrikahar** [comment]
> Has this been tested against any DBs? In Postgres and related DBs, this will throw an error since the grouping is done based on the precision (year, month, week...etc.,) and the dateField used in the sort query is not part of the group query.

Personally haven't testing this per se, but I do believe this is also what was happening over here: https://github.com/directus/directus/pull/8243#discussion_r757295243.

**Comment by jaycammarano** [comment]
> Has this been tested against any DBs? In Postgres and related DBs, this will throw an error since the grouping is done based on the precision (year, month, week...etc.,) and the dateField used in the sort query is not part of the group query.

Sorry. I meant to get some tests going for this and totally lost track of the PR. Working on it now.

**Comment by jaycammarano** [comment]
> Has this been tested against any DBs? In Postgres and related DBs, this will throw an error since the grouping is done based on the precision (year, month, week...etc.,) and the dateField used in the sort query is not part of the group query.

Do you mind testing this with your dataset? I'm not entirely sure I had the same problem with the dataset as you.

File: app/src/panels/time-series/time-series.vue [hunk 1]
```diff
@@ -228,7 +232,12 @@ export default defineComponent({
 				return `${year}-${month}-${day}T${hour}:${minute}:${second}`;
 
 				function padZero(value: number) {
-					return String(value).padStart(2, '0');
+					const intAndDecimal = String(value).split('.');
+
+					if (intAndDecimal.length > 1) {
+						return `${String(intAndDecimal[0]).padStart(2, '0')}.${intAndDecimal[1]}`;
+					}
+					return String(intAndDecimal[0]).padStart(2, '0');
```
Comment (rijkvanzanten) [inline app/src/panels/time-series/time-series.vue:240]: padZero isn't used in the function above. Does this solve another issue? If so, lets move it into another PR 👍🏻

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: app/src/panels/time-series/time-series.vue (hunk 1 above)
Comment (jaycammarano) [reply app/src/panels/time-series/time-series.vue:240]: It was an issue where the date strings were returning "invalid date" because of the truncating. I ran into it when I was doing this fix. It might not be an actual issue in production though.

**Comment by rijkvanzanten** [comment]
Superseded by https://github.com/directus/directus/pull/14096
